parser/parser.go

```diff
@@ -110,3 +122,25 @@ func (p *parser) many(start lexer.Type, end lexer.Type, cb func()) {
 	}
 	p.next()
 }
+
+func (p *parser) some(start lexer.Type, end lexer.Type, cb func()) {
+	p.mark()
```

Comment:
I think you can can avoid this by simply sending an error:
```
func (p *parser) some(start lexer.Type, end lexer.Type, description string, cb func()) {
	hasDef := p.skip(start)
	if !hasDef {
		return
	}

	called := false
	for p.peek().Kind != end && p.err == nil {
		called = true
		cb()
	}

	if !called {
		p.error(p.peek(), "expected at least one %s, found %s", description, p.peek().Kind.String())
		return
	}

	p.next()
}
```

and update the test like:

```go
		t.Run("can't read empty array", func(t *testing.T) {
			p := newParser("[]")

			var arr []string
			p.some(lexer.BracketL, lexer.BracketR, "things", func() {
				arr = append(arr, p.next().Value)
			})
			require.EqualError(t, p.err, "input.graphql:1: expected at least one things, found ]")
			require.Equal(t, []string(nil), arr)
		})
```

The way you have it it wont currently error, and the error will fire when trying to parse the next token which will be confusing to the user.